### PR TITLE
Tethering

### PR DESCRIPTION
**Pull request by ghost**
Enable tethering on comma devices via network manager.

### CODE REVIEW

File: selfdrive/ui/qt/offroad/wifiManager.cc [hunk 1]
```diff
@@ -54,6 +56,16 @@ WifiManager::WifiManager(){
     raw_adapter_state = get_response<uint>(response);
     change(raw_adapter_state, 0, 0);
   }
+  // Compute tethering ssid as "Weedle" + first 4 characters of serial number of a device
+  tethering_ssid = "Weedle";
+  QString path = "DongleId";
+  std::vector<char> bytes = Params().read_db_bytes(path.toStdString().c_str());
```
Comment (pd0wm) [inline selfdrive/ui/qt/offroad/wifiManager.cc:62]: Just hardcode as `Params().read_db_bytes("DongleId");`

File: selfdrive/ui/qt/offroad/wifiManager.cc [hunk 2]
```diff
@@ -54,6 +56,16 @@ WifiManager::WifiManager(){
...
+    
+  if (bytes.size() >= 4){
```
Comment (pd0wm) [inline selfdrive/ui/qt/offroad/wifiManager.cc:64]: Append without loop.

File: selfdrive/ui/qt/offroad/wifiManager.cc [hunk 3]
```diff
@@ -294,3 +305,44 @@ void WifiManager::change(unsigned int new_state,unsigned int previous_state,unsi
     connecting_to_network="";
   }
 }
+
+
+//Functions for tethering 
+
+void WifiManager::enableTethering(){
+  // Disconnect from active access_point
```
Comment (pd0wm) [inline selfdrive/ui/qt/offroad/wifiManager.cc:313]: Isn't this code also needed when changing networks. Abstract into function.

File: selfdrive/ui/qt/offroad/wifiManager.cc [hunk 4]
```diff
@@ -294,3 +305,44 @@ void WifiManager::change(unsigned int new_state,unsigned int previous_state,unsi
...
+
+  connection["802-11-wireless"]["band"] = "bg";
+  connection["802-11-wireless"]["mode"] = "ap";
+  connection["802-11-wireless"]["ssid"] = tethering_ssid.toUtf8();
+
+  connection["802-11-wireless-security"]["key-mgmt"] = "wpa-psk";
+  connection["802-11-wireless-security"]["auth-alg"] = "open";
+  connection["802-11-wireless-security"]["psk"] = "swagswagcomma";
+
+  connection["ipv4"]["method"] = "shared";
```
Comment (pd0wm) [inline selfdrive/ui/qt/offroad/wifiManager.cc:334]: Where is the IP range? That range is important.

**Review by pd0wm — DISMISSED** [review]

File: selfdrive/ui/qt/offroad/wifiManager.cc (hunk 1 above)
Comment (adeebshihadeh) [reply selfdrive/ui/qt/offroad/wifiManager.cc:62]: also use get, we're trying to move everything to the python params API

**Pull request merged**